validator/rpc/accounts.go

```diff
@@ -179,16 +189,27 @@ func (s *Server) DeleteAccounts(
 func (s *Server) VoluntaryExit(
 	ctx context.Context, req *pb.VoluntaryExitRequest,
 ) (*pb.VoluntaryExitResponse, error) {
+	var km keymanager.IKeymanager
+	if s.validatorService == nil {
+		return nil, status.Error(codes.FailedPrecondition, "Validator service not yet initialized")
+	}
 	if len(req.PublicKeys) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "No public keys specified to delete")
 	}
-	if s.wallet == nil || s.keymanager == nil {
+	if s.wallet == nil {
 		return nil, status.Error(codes.FailedPrecondition, "No wallet found")
 	}
-	if s.wallet.KeymanagerKind() != keymanager.Imported && s.wallet.KeymanagerKind() != keymanager.Derived {
-		return nil, status.Error(
-			codes.FailedPrecondition, "Only Imported or Derived wallets can submit voluntary exits",
-		)
+	ikm, err := s.validatorService.Keymanager()
+	if err != nil {
+		return nil, err
+	}
+	switch ikm := ikm.(type) {
+	case *imported.Keymanager:
+		km = ikm
+	case *derived.Keymanager:
+		km = ikm
+	default:
+		return nil, status.Error(codes.FailedPrecondition, "Only Imported or Derived wallets can delete accounts")
```

Comment:
I dont understand this. Why can't a remote signer perform a voluntary exit?

Reply:
some old code threw me off I don't understand why the current remote signer can't voluntarily exit. i made change to remove this restriction